bytearray_operations.go

```diff
@@ -44,6 +44,83 @@ func LeftShift(data []byte, shift uint64) []byte {
 	return result
 }
 
+// Mask apply AND mask to byte array
+func Mask(data, mask []byte) []byte {
+	var dataLength = len(data)
+	result := make([]byte, dataLength)
+	if dataLength < 1 {
+		return result
+	}
+	copy(result, data)
+
+	var maskLength = len(mask)
+	var operationLength = maskLength
+	// If mask is longer than data, keep operation to data length
+	if maskLength > dataLength {
+		operationLength = dataLength
+	}
+	for i := 0; i < operationLength; i++ {
+		result[i] = data[i] & mask[i]
```

Comment:
Since `result[i]` is already equals to `data[i]` (because of the array copy). You can use `r &= m` instead of `r = d & m`